src/main/webapp/src/components/youTubeRecommendations.js

```diff
@@ -41,24 +42,66 @@ class YouTubeRecommendations extends React.Component {
     };
   }
 
-  componentDidMount() {
-    this.setState({ isLoading: true });
-
-    axios
-      .get(`/youtube?query=${this.props.q}`)
-      .then((result) => result.data)
-      .then((videoIds) => {
-        this.setState({
-          videoIds: videoIds,
-          isLoading: false,
-        })
-      })
-      .catch((error) =>
+  /**
+   * Load component if entity component still loading.
+   * Show error if entity component rendered with error.
+   * Find videos based on the most salient word if 
+   * entities were found.
+   * @param {Object} entityState 
+   */
+  getVideos = (entityState) => {
+    if (entityState != undefined) {
+      if (entityState.errorMsg) {
         this.setState({
-          error,
+          error: new Error(entityState.errorMsg),
           isLoading: false,
-        })
-      );
+        });
+      } else if (!entityState.isLoading) {
+        if (entityState.entityAnalysisInfo.length == 0) {
+          this.setState({
+            error: new Error(
+              "No YouTube videos were found, because no entities were found"
+            ),
+            isLoading: false,
+          });
+        } else {
+          axios
+            .get(
+              `/youtube?query=${entityState.entityAnalysisInfo[0].name}`
+            )
+            .then((result) => result.data)
+            .then((videoIds) => {
+              this.setState({
+                videoIds: videoIds,
+                isLoading: false,
+              });
+            })
+            .catch((error) =>
+              this.setState({
+                error,
+                isLoading: false,
+              })
+            );
+        }
+      }
+    } else {
+      this.setState({ isLoading: true });
+    }
```

Comment:
For readability, I would switch the two conditions here, and it is usually better to have positive conditions if possible. So I would do
```
else if (entityState.isLoading) {
...
} else {
...
}
```